src/webhook/webhook.go

```diff
@@ -0,0 +1,72 @@
+package webhook
+
+import (
+	"github.com/goharbor/harbor/src/common/utils/log"
+	cfg "github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/webhook/config"
+	"github.com/goharbor/harbor/src/webhook/hook"
+	"github.com/goharbor/harbor/src/webhook/job"
+	"github.com/goharbor/harbor/src/webhook/job/controller"
+	"github.com/goharbor/harbor/src/webhook/model"
+	"github.com/goharbor/harbor/src/webhook/policy"
+)
+
+var (
+	// PolicyCtl is a global webhook policy controller
+	PolicyCtl policy.Controller
+
+	// JobCtl is a webhook job controller
+	JobCtl job.Controller
+
+	// HookManager is a hook manager
+	HookManager hook.Manager
+
+	// SupportedHookTypes is a map store support webhook type, eg. pushImage, pullImage etc
+	SupportedHookTypes map[string]int
+
+	// SupportedSendTypes is a map store webhook send type, eg. HTTP, Email etc
+	SupportedSendTypes map[string]int
```

Comment:
Webhook  and Email should be two different handlers.  i.e. you cannot send webhook via Email

Reply:
> ook and Email should be two different handlers. i.e. you

I think Email is a send method, just like `http`

Reply:
I think you have a misunderstanding...  Email is NOT webhook:
https://en.wikipedia.org/wiki/Webhook

Reply:
> I think you have a misunderstanding... Email is NOT webhook:
> https://en.wikipedia.org/wiki/Webhook

code fixed. change all `webhook` -> `notification`